compiler/passes/normalize.cpp

```diff
@@ -1744,6 +1764,14 @@ static void change_method_into_constructor(FnSymbol* fn) {
     }
   }
 
+  if (ct->instantiationStyle == DEFINES_INITIALIZER) {
+    ArgSymbol* meme = new ArgSymbol(INTENT_BLANK, "meme", ct, NULL,
```

Comment:
Scope resolve might not know the difference between a class and a record. We wouldn't want a class instance pointer to come in with ref intent. Also, RE INIT REF, I think it's sufficient to treat the meme argument specially. I don't think it's used in enough other situations that we need FLAG_INIT_REF. One could argue that the retArg argument has a similar property but I don't think making that property part of the intent would help either of these cases.